planner/core/rule_partition_processor.go

```diff
@@ -224,73 +224,251 @@ func (s *partitionProcessor) processHashPartition(ds *DataSource, pi *model.Part
 	return tableDual, nil
 }
 
-func (s *partitionProcessor) findUsedListPartitions(ctx sessionctx.Context, tbl table.Table, partitionNames []model.CIStr,
-	conds []expression.Expression) ([]int, error) {
-	pi := tbl.Meta().Partition
-	partExpr, err := tbl.(partitionTable).PartitionExpr()
-	if err != nil {
-		return nil, err
-	}
+// listPartitionPruner uses to prune partition for list partition.
+type listPartitionPruner struct {
+	*partitionProcessor
+	ctx             sessionctx.Context
+	pi              *model.PartitionInfo
```

Comment:
Why listPartitionPruner stores a `*model.PartitionInfo`?
After `alter table add partition ...`, is this field updated?

Reply:
`listPartitionPruner` not a meta-information that store in `infoschema`. It is just a `prune-helper`, put the related list pruner function as a method of `listPartitionPruner` to avoid too many function input parameters.